### PR TITLE
Feature/add ofeli

### PR DESCRIPTION
**Pull request by Kr4is**
Specify library name and version:  **ofeli/4.1.2**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Kr4is** [comment]
This PR solves #8333

**Comment by Kr4is** [comment]
This PR resolve #8333

File: recipes/ofeli/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        if not self.settings.compiler.cppstd:
+            self._autotools.cppstd_flag = "-std=c++11"
+        self._autotools.configure(args=["--enable-release"])
+        return self._autotools
+
+    def configure(self):
+        if self.settings.os == "Linux":
+            self.settings.compiler.libcxx = "libstdc++11"
```
Comment (ericLemanissier) [inline recipes/ofeli/all/conanfile.py:55]:
```suggestion
```
Is this really needed ? Is it even allowed to modify the settings from a recipe ? 
`libstdc++` is also compatible with c++11

File: recipes/ofeli/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def config_options(self):
+        self.settings.compiler.libcxx = "libstdc++11"
```
Comment (ericLemanissier) [inline recipes/ofeli/all/test_package/conanfile.py:10]:
```suggestion
```
Same here

File: recipes/ofeli/all/conanfile.py (hunk 1 above)
Comment (Kr4is) [reply recipes/ofeli/all/conanfile.py:55]: I can try to put it as an option in the configuration, so that the user can choose between libstdc++ and libstdc++11. Is this okay?

Comment (ericLemanissier) [reply recipes/ofeli/all/conanfile.py:55]: Why not using the regular way to define settings: `-s` command line argument, or in the profile?

Comment (Kr4is) [reply recipes/ofeli/all/conanfile.py:55]: i have tried to avoid use libstdc++11 at the beginning but it did not work, so y add it at this commit: https://github.com/conan-io/conan-center-index/pull/8375/commits/ba0b21a1aaa7c4362fa4b301f7ec07cd5e407412, It is possible that the library needs that flag?

Comment (Kr4is) [reply recipes/ofeli/all/conanfile.py:55]: Hello again, reviewing the history, the main problem is that with the gcc compiler in version 5, we need this configuration to compile correctly.

Comment (ericLemanissier) [reply recipes/ofeli/all/conanfile.py:55]: The problem with gcc5 was actually solved in 9ca059b1237f34e693bb344fe85bbf28b5c3d503

File: recipes/ofeli/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        if not self.settings.compiler.cppstd:
+            self._autotools.cppstd_flag = "-std=c++11"
```
Comment (ericLemanissier) [inline recipes/ofeli/all/conanfile.py:49]:
```suggestion
```
This should not be needed either

**Review by ericLemanissier — CHANGES_REQUESTED** [review]

File: recipes/ofeli/all/conanfile.py (hunk 1 above)
Comment (Kr4is) [reply recipes/ofeli/all/conanfile.py]: I just made the changes, I am waiting for the CI to finish. Thanks for your response!

File: recipes/ofeli/all/test_package/conanfile.py (hunk 2 above)
Comment (Kr4is) [reply recipes/ofeli/all/test_package/conanfile.py]: I just made the changes, I am waiting for the CI to finish. Thanks for your response!

File: recipes/ofeli/all/conanfile.py (hunk 3 above)
Comment (Kr4is) [reply recipes/ofeli/all/conanfile.py:47]: I just made the changes, I am waiting for the CI to finish. Thanks for your response!

File: recipes/ofeli/all/conanfile.py (hunk 1 above)
Comment (Kr4is) [reply recipes/ofeli/all/conanfile.py]: Hi again, with the changes you commented, it seems that it fails when executing the test example with gcc 8 :S Any idea why the stdc ++ 11 flag might work?

**Comment by ericLemanissier** [comment]
@madebr the test fails for `compiler.libcxx=libstdc++` but not for `compiler.libcxx=libstdc++11`, conan's AutoToolsBuildEnvironment should honor the libcxx setting right ? Do you have an idea ?

File: recipes/ofeli/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        self._autotools.configure(args=["--enable-release"])
```
Comment (ericLemanissier) [inline recipes/ofeli/all/conanfile.py:48]:
```suggestion
        self._autotools.configure(args=["--enable-%s" % ("release" if self.settings.build_type == "Release" else "debug")])
```

File: recipes/ofeli/all/conanfile.py [hunk 5]
```diff
@@ -34,8 +34,6 @@ def validate(self):
         if self.settings.compiler != "gcc":
             raise ConanInvalidConfiguration(
                 "Ofeli is just supported for GCC")
-        if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, 11)
```
Comment (ericLemanissier) [inline recipes/ofeli/all/conanfile.py]: why did you remove this ?

Comment (Kr4is) [reply recipes/ofeli/all/conanfile.py]: i remove it to avoid force cppstd11, but i can add it!

File: recipes/ofeli/all/conanfile.py (hunk 4 above)
Comment (Kr4is) [reply recipes/ofeli/all/conanfile.py]: great! i will add this change! thanks

File: recipes/ofeli/all/conanfile.py (hunk 3 above)
Comment (uilianries) [reply recipes/ofeli/all/conanfile.py:49]: it's still present

File: recipes/ofeli/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+required_conan_version = ">=1.40.0"
+
+
+class OfeliConan(ConanFile):
+    name = "ofeli"
+    description = "An Object Finite Element Library"
+    topics = ("ofeli", "finite element", "finite element library",
+              "finite element analysis", "finite element solver")
+    license = "GNU Lesser General Public License (LGPL)."
```
Comment (uilianries) [inline recipes/ofeli/all/conanfile.py:13]:
```suggestion
    license = "LGPL-3.0-or-later"
```

Comment (Kr4is) [reply recipes/ofeli/all/conanfile.py]: Done!

File: recipes/ofeli/all/conanfile.py (hunk 3 above)
Comment (Kr4is) [reply recipes/ofeli/all/conanfile.py]: I have removed it but i don't know if it is needed for work correctly

**Review by uilianries — APPROVED** [review]

File: recipes/ofeli/all/test_package/conanfile.py (hunk 2 above)
Comment (prince-chrismc) [reply recipes/ofeli/all/test_package/conanfile.py:10]: Bump

Comment (Kr4is) [reply recipes/ofeli/all/test_package/conanfile.py:10]: Without this config option the test don't works, is not correctly have this here? What i could do for solve this? greetings

Comment (prince-chrismc) [reply recipes/ofeli/all/test_package/conanfile.py:10]: Thats a good sign the binaries are not being compiled with the expected conan settings.

It seems the project hardcodes it to this https://github.com/rtouzani/ofeli/search?l=Makefile&q=std (i didnt check all 17 pages but I assume it's everywhere)

Since you already limited this to gcc [here](https://github.com/conan-io/conan-center-index/pull/8375/files#diff-1707c887febaab51f38cd2d4e92ee705c47eeabbe5fc8a6e5fe80c8578beb9afR34-R36)
 you can also limit the `libcxx` setting as well

File: recipes/ofeli/all/test_package/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.4)
+project(test_package LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_STANDARD 11)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```
Comment (ericLemanissier) [inline recipes/ofeli/all/test_package/CMakeLists.txt:8]:
```suggestion
conan_basic_setup(TARGETS)
```

File: recipes/ofeli/all/test_package/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (ericLemanissier) [inline recipes/ofeli/all/test_package/CMakeLists.txt:11]:
```suggestion
target_link_libraries(${PROJECT_NAME} CONAN_PKG::ofeli)
```

File: recipes/ofeli/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(
+                "Ofeli is just supported for Linux")
+        if self.settings.compiler != "gcc":
+            raise ConanInvalidConfiguration(
+                "Ofeli is just supported for GCC")
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
```
Comment (ericLemanissier) [inline recipes/ofeli/all/conanfile.py:39]:
```suggestion
    if self.settings.compiler.libcxx != "libstdc++11":
        raise ConanInvalidConfiguration("Ofeli supports only libstdc++'s new ABI")

```

File: recipes/ofeli/all/test_package/conanfile.py (hunk 2 above)
Comment (ericLemanissier) [reply recipes/ofeli/all/test_package/conanfile.py:10]: Please remove this , and apply my other suggestions. Ofeli actually ignores `CPPFLAGS` provided by conan : https://github.com/rtouzani/ofeli/blob/524b2cc3ed51fa984c9a10d685828f41c3d7a2e4/configure.ac#L47 , which makes it use the new libstdc++ abi by default

**Review by ericLemanissier — CHANGES_REQUESTED** [review]

File: recipes/ofeli/all/test_package/CMakeLists.txt (hunk 7 above)
Comment (Kr4is) [reply recipes/ofeli/all/test_package/CMakeLists.txt]: Done!

File: recipes/ofeli/all/test_package/CMakeLists.txt (hunk 8 above)
Comment (Kr4is) [reply recipes/ofeli/all/test_package/CMakeLists.txt]: Done!

**Review by ericLemanissier — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/ofeli/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(
+                "Ofeli is just supported for Linux")
+        if self.settings.compiler != "gcc":
+            raise ConanInvalidConfiguration(
+                "Ofeli is just supported for GCC")
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+        if self.settings.compiler.libcxx != "libstdc++11":
```
Comment (uilianries) [inline recipes/ofeli/all/conanfile.py:39]: Which scenario it doesn't work? I'm able to with gcc and libstdc++ without any error.

Comment (ericLemanissier) [reply recipes/ofeli/all/conanfile.py:39]: cf https://github.com/conan-io/conan-center-index/pull/8375#discussion_r777419520
The failure is the one happening in https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/8375/8-configs/linux-gcc/ofeli/4.1.2//summary.json for example
This is expected, because the upstream project ovewrites CPPFLAGS passed by conan, so it does not respect the profile regarding the cpp standard used, or libstdc++'s abi, so it ends up using the deafult, which is libstdc++11

Comment (uilianries) [reply recipes/ofeli/all/conanfile.py:39]: Thank you for these details!

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/ofeli/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                  src=os.path.join(self._source_subfolder, "include"))
+        self.copy("*libofeli.a", dst="lib",
+                  src=os.path.join(self._source_subfolder, "src"))
+        self.copy("*.md", dst="res",
+                  src=os.path.join(self._source_subfolder, "material"))
+        self.copy("COPYING", dst="licenses", src=self._doc_folder)
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "Ofeli"
+        self.cpp_info.names["cmake_find_package_multi"] = "Ofeli"
```
Comment (SpaceIm) [inline recipes/ofeli/all/conanfile.py:72]: where does it come from?

Comment (uilianries) [reply recipes/ofeli/all/conanfile.py:72]: Good catch. Master branch offers cmake support, but still is all lowercase.

Comment (uilianries) [reply recipes/ofeli/all/conanfile.py:72]: Fix available on https://github.com/conan-io/conan-center-index/pull/9050
